test/parallel/test-fs-readdir-types.js

```diff
@@ -49,6 +49,12 @@ function assertDirents(dirents) {
 // Check the readdir Sync version
 assertDirents(fs.readdirSync(readdirDir, { withFileTypes: true }));
 
+fs.readdir(__filename, {
+  withFileTypes: true
+}, common.mustCall((err) => {
+  assert.ok(err);
+}));
```

Comment:
As a suggestion:

To test the actual error it would be best to rewrite this to e.g.,:

```js
assert.throws(
  () => fs.readdir(__filename, { withFileTypes: true },
  {
    message: 'FOOBAR',
    name: 'ErrorName',
    ...
  }
);
```

Reply:
@BridgeAR the error is actually passed within the `fs.readdir` callback, how about:

```js
fs.readdir(__filename, {
  withFileTypes: true
}, common.mustCall((err) => {
  assert.throws(
    () => throw err,
    {
      code: 'FOO',
      name: 'TypeError [FOO]',
      ...
    }
  );
}));
```

Reply:
Right, sorry, I made a mistake when writing the example. I just updated it.